### PR TITLE
fix package_data in docker images

### LINKED ISSUE #4593: jupyterhub/jupyterhub image fails (`alembic.ini` missing) by default - is it a bug?
I'm not sure what the intented behavior of the jupyterhub/jupyterhub image is, but currently it fails to startup by default without additional mounted configuration files etc via a `docker run docker.io/jupyterhub/jupyterhub:main` command.

```
docker run docker.io/jupyterhub/jupyterhub:main

[I 2023-10-02 10:16:39.360 JupyterHub app:3076] Running JupyterHub version 5.0.0.dev
[I 2023-10-02 10:16:39.360 JupyterHub app:3106] Using Authenticator: jupyterhub.auth.PAMAuthenticator-5.0.0.dev
[I 2023-10-02 10:16:39.361 JupyterHub app:3106] Using Spawner: jupyterhub.spawner.LocalProcessSpawner-5.0.0.dev
[I 2023-10-02 10:16:39.361 JupyterHub app:3106] Using Proxy: jupyterhub.proxy.ConfigurableHTTPProxy-5.0.0.dev
[I 2023-10-02 10:16:39.384 JupyterHub app:1768] Writing cookie_secret to /srv/jupyterhub/jupyterhub_cookie_secret
[E 2023-10-02 10:16:39.425 JupyterHub app:3639]
    Traceback (most recent call last):
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 3636, in launch_instance_async
        await self.initialize(argv)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 3118, in initialize
        self.init_db()
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/app.py", line 1897, in init_db
        self.session_factory = orm.new_session_factory(
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/orm.py", line 1161, in new_session_factory
        check_db_revision(engine)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/orm.py", line 1052, in check_db_revision
        with _temp_alembic_ini(engine_url) as ini:
      File "/usr/lib/python3.10/contextlib.py", line 135, in __enter__
        return next(self.gen)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/dbutil.py", line 71, in _temp_alembic_ini
        write_alembic_ini(alembic_ini, db_url)
      File "/usr/local/lib/python3.10/dist-packages/jupyterhub/dbutil.py", line 34, in write_alembic_ini
        with open(ALEMBIC_INI_TEMPLATE_PATH) as f:
    FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/lib/python3.10/dist-packages/jupyterhub/alembic.ini'
```

### PR DESCRIPTION
**Pull request by minrk**
- verify package_data and data_files installation in docker build stage
- include .git in build stage, setuptools_scm section to ensure setuptools_scm gets its file metadata correct

closes #4593

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -67,6 +68,11 @@ ARG PIP_CACHE_DIR=/tmp/pip-cache
 RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
     python3 -m build --wheel
 
+# verify installed files
+RUN --mount=type=cache,target=${PIP_CACHE_DIR} \
+    python3 -m pip install ./dist/*.whl \
+ && cd ci \
+ && python3 check_installed_data.py
```
Comment (minrk) [inline Dockerfile:75]: This passing verifies that the change does indeed fix the missing files

**Review by consideRatio — APPROVED** [review]

**Review by consideRatio — APPROVED** [review]
LGTM!

**Comment by minrk** [comment]
@meeseeksdev backport to 4.x

**Comment by minrk** [comment]
oops, doesn't make sense to backport this one, since it was a regression in 5.0 changes.

**Pull request merged**